### PR TITLE
[SYM-4798] Update S3 Log Destination example to use runtime-connector module 2.x

### PR DESCRIPTION
**Pull request by Arianna2028**
## Description

- Updates the `basic/s3_log_destination` example to use version 2.x of the [runtime-connector](https://registry.terraform.io/modules/symopsio/runtime-connector/aws/latest) module.
- Updates the usages of `main` to reference the `local.environment_name` variable instead, for consistency.
- Updates the Sym Terraform Provider to v3 and the `sym_flow.implementation` to be file contents, which is required in v3.
- Fixes a bug where logs would not be sent because the `sym_log_destination` was not set in the `sym_environment`.

## Testing

Tested using the alpha version of the runtime-connector module and the following `override.tf`:
```hcl
provider "sym" {
  org = "sym"
}

module "runtime_connector" {
  source  = "symopsio/runtime-connector/aws"
  version = "2.0.0-alpha"

  environment_name = local.environment_name
  sym_account_id = "906191524257"
}
```

### CODE REVIEW

**Comment by Arianna2028** [comment]
Current dependencies on/for this PR:
* main
  * **PR #71** <a href="https://app.graphite.dev/github/pr/symopsio/examples/71" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/symopsio/examples/71?utm_source=stack-comment).

File: basic/s3_log_destination/connectors.tf [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  version = "~> 2.0"
+
+  environment = local.environment_name
+}
+
+# The kinesis_firehose_access module generates an AWS IAM Policy that grants permissions to publish to the given AWS Kinesis Firehose.
+# Those permissions will be granted to the Runtime Connector IAM Role so that the Sym Runtime can publish to the Kinesis Firehose.
+module "kinesis_firehose_access" {
+  source  = "symopsio/kinesis-firehose-addon/aws"
+  version = ">= 1.1.0"
```
Comment (llam15) [inline basic/s3_log_destination/connectors.tf:21]:
```suggestion
  version = "~> 1.1"
```

Same thought here about pinning a major version?

File: basic/s3_log_destination/environment.tf [hunk 2]
```diff
@@ -11,6 +11,7 @@ provider "sym" {
 resource "sym_environment" "this" {
   name            = local.environment_name
   error_logger_id = sym_error_logger.slack.id
+  log_destination_ids = [sym_log_destination.s3_firehose.id]
```
Comment (llam15) [inline basic/s3_log_destination/environment.tf:14]: Good catch 😅

File: basic/s3_log_destination/s3_log_destination.tf [hunk 3]
```diff
@@ -21,7 +21,7 @@ resource "aws_kinesis_firehose_delivery_stream" "sym_logs" {
   }
 
   tags = {
-    # This SymEnv tag is required and MUST match the SymEnv tag in the 
+    # This SymEnv tag is required and MUST match the SymEnv tag in the
     # aws_iam_policy.secrets_manager_access in your `secrets.tf` file
```
Comment (llam15) [inline basic/s3_log_destination/s3_log_destination.tf:25]: I think this comment should be referencing the `kinesis_firehose_connector` module?

File: basic/s3_log_destination/connectors.tf [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+  environment = local.environment_name
```
Comment (llam15) [inline basic/s3_log_destination/connectors.tf:23]: nit: `terraform fmt`?

**Review by llam15 — APPROVED** [review]

**Pull request merged**